pkg/codec/mmal/mmal.go

```diff
@@ -91,12 +91,14 @@ func (e *encoder) Read() ([]byte, func(), error) {
 	return encoded, func() {}, err
 }
 
-func (e *encoder) SetBitRate(b int) error {
-	panic("SetBitRate is not implemented")
-}
+// TODO: Implement Bit rate control
+// var _ codec.BitRateController = (*encoder)(nil)
+
+// TODO: Implement Key Frame control
+//var _ codec.KeyFrameController = (*encoder)(nil)
```

Comment:
It may be nice to be in `_test.go` file to clarify it is interface type check.

Reply:
You mean moving the type check lines in a `_test.go` ?

Reply:
Yes, I think it's better to separate testing purpose code which is not used in the product into `*_test.go` files.

Reply:
Ok, I have written real test instead of the hacky type check.

I think it is in fact better since it test the returned value of `Controller` is implementing the interfaces instead of hardcoded `encoder` type. This way, I don't make any assumption on how this is implemented.